CMSgov/easi-app
# [WIP] EASI-2513 - TRB Admin Notes

# EASI-2513

WIP pull request, mainly to copy down manual tests.

## Changes and Description

- Introduce TRB Admin notes
    - Add SQL migration to create table for them
    - Create store methods for creating and fetching them
    - Add them to GraphQL schema
    - Add resolvers for creating and fetching them

## How to test this change

1. Spin up the backend and open the GraphQL playground.
2. Create a TRB request, noting the ID for later:
```gql
mutation {
  createTRBRequest(requestType: NEED_HELP) {
    id
    type
  }
}
```
3. Query for the notes on the request - there should be no notes (since none have been created), but no errors. (Save this query, it'll be used again)
```gql
query {
  trbRequest(id:"trbRequestIdFromAbove") {
    adminNotes {
      id
      trbRequestId
      category
      noteText
      createdBy
      createdAt
      modifiedBy
      modifiedAt
    }
  }
}
```
4. Create an admin note.
```gql
mutation {
  createTRBAdminNote(
    input: {
      trbRequestId: "trbRequestIdFromAbove"
      category: GENERAL_REQUEST
      noteText: "Needs some more polish"
    }
  ) {
    id
    trbRequestId
    category
    noteText
    createdBy
    createdAt
    modifiedBy
    modifiedAt
  }
}
```
5. Run the query from step 3 again. You should see the created note, with the correct category and text.
6. Create a second admin note.
```gql
mutation {
  createTRBAdminNote(
    input: {
      trbRequestId: "trbRequestIdFromAbove"
      category: ADVICE_LETTER
      noteText: "Maybe soften the criticism a bit"
    }
  ) {
    id
    trbRequestId
    category
    noteText
    createdBy
    createdAt
    modifiedBy
    modifiedAt
  }
}
```
7. Run the query from step 3 again. You should see both created notes, with distinct data for each note.

## Notes
- Similar to https://github.com/CMSgov/easi-app/pull/1865, I didn't add any unit tests for the resolver functions due to their simplicity.
